# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmBuiltins.h
# Commit: 44b0c0bed8a6
# Full Hash: 44b0c0bed8a6b4debb950e7e1c69190883c65d60
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2021-10-18 16:09:29
# Regressor Bug: 1727084
# File Overlap Count: 1
# Description:
#   Bug 1727084 - Memory64 - Atomic-rmw baseline operations. r=yury
#   
#   Atomic ADD, SUB, AND, OR, XOR, XCHG, CMPXCHG, as well as WAIT and
#   NOTIFY for the baseline compiler.
#   
# ==============================================================================

diff -r fb02656bb788 -r 44b0c0bed8a6 js/src/wasm/WasmBuiltins.h
--- a/js/src/wasm/WasmBuiltins.h	Mon Oct 18 09:58:15 2021 +0000
+++ b/js/src/wasm/WasmBuiltins.h	Mon Oct 18 09:58:15 2021 +0000
@@ -91,9 +91,12 @@
   MemoryGrowM64,
   MemorySizeM32,
   MemorySizeM64,
-  WaitI32,
-  WaitI64,
-  Wake,
+  WaitI32M32,
+  WaitI32M64,
+  WaitI64M32,
+  WaitI64M64,
+  WakeM32,
+  WakeM64,
   MemCopyM32,
   MemCopySharedM32,
   MemCopyM64,
@@ -214,9 +217,12 @@
 extern const SymbolicAddressSignature SASigMemoryGrowM64;
 extern const SymbolicAddressSignature SASigMemorySizeM32;
 extern const SymbolicAddressSignature SASigMemorySizeM64;
-extern const SymbolicAddressSignature SASigWaitI32;
-extern const SymbolicAddressSignature SASigWaitI64;
-extern const SymbolicAddressSignature SASigWake;
+extern const SymbolicAddressSignature SASigWaitI32M32;
+extern const SymbolicAddressSignature SASigWaitI32M64;
+extern const SymbolicAddressSignature SASigWaitI64M32;
+extern const SymbolicAddressSignature SASigWaitI64M64;
+extern const SymbolicAddressSignature SASigWakeM32;
+extern const SymbolicAddressSignature SASigWakeM64;
 extern const SymbolicAddressSignature SASigMemCopyM32;
 extern const SymbolicAddressSignature SASigMemCopySharedM32;
 extern const SymbolicAddressSignature SASigMemCopyM64;